toolchain/lower/testdata/basics/type_values.carbon

```diff
@@ -15,10 +15,12 @@ fn F64() -> type {
 // CHECK:STDOUT: ; ModuleID = 'type_values.carbon'
 // CHECK:STDOUT: source_filename = "type_values.carbon"
 // CHECK:STDOUT:
-// CHECK:STDOUT: define void @I32() {
-// CHECK:STDOUT:   ret void
+// CHECK:STDOUT: %type = type {}
+// CHECK:STDOUT:
+// CHECK:STDOUT: define %type @I32() {
+// CHECK:STDOUT:   ret %type zeroinitializer
```

Comment:
This seems a bit surprising, if likely harmless. Mostly flagging in case you weren't expecting it or want to leave a TODO.

Reply:
It's not ideal, but it is expected: we're switching from the value representation of `type` being empty to it being `type`-by-copy. I think that's right at the SemIR level: for compile-time constant evaluation, the result really is not "none". But the toolchain doesn't have any separate layout optimizations for struct beyond the ones that are done in SemIR, and currently does a very naive conversion from SemIR types to LLVM types, so changing the SemIR side of things loses the lowering optimization. (This also comes up with things like `(i32, (), i32)`, which is lowered to `{i32, {}, i32}` -- both before and after this PR -- but I think should really be just `{i32, i32}`.)

(And it turns out to be ... really annoying ... to preserve the current SemIR value representation due to the very special treatment of `TypeType` as a builtin that has no entry in or types table. I looked into removing that special treatment, but it's going to take more changes that I was comfortable putting into this PR.)

Added a TODO to improve the lowering.